client/client.go

```diff
@@ -12,34 +12,49 @@ import (
 )
 
 var (
-	app   = kingpin.New("client", "send messages to a server")
-	host  = app.Flag("host", "host to connect to").Required().String()
-	send  = app.Flag("message", "send a request to server").Required().Strings()
-	token = app.Flag("token", "authentication token").Required().String()
+	app = kingpin.New("client", "connect to server")
+
+	send    = app.Command("send", "send message to server")
+	host    = send.Flag("host", "host to connect to").Required().String()
+	message = send.Flag("message", "send a request to server").Required().Strings()
+	token   = app.Flag("token", "authentication token").Required().String()
 )
 
 func main() {
-	kingpin.MustParse(app.Parse(os.Args[1:]))
-	message := strings.Join(*send, " ")
-	reader := strings.NewReader(message)
-
-	req, err := http.NewRequest("POST", *host+"/echo", reader)
-
-	// add token header to the req
-	req.Header.Add("Token", *token)
-
-	client := &http.Client{}
-	resp, err := client.Do(req)
-
-	if err != nil {
-		log.Println(err)
-	}
-
-	defer resp.Body.Close()
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	serverOutput := string(bodyBytes)
-	if serverOutput != message {
-		fmt.Println(*host + " did not echo your request, oops! got:")
+	switch kingpin.MustParse(app.Parse(os.Args[1:])) {
+
+	case send.FullCommand():
```

Comment:
The reason I didn't have a default case in client.go was because the flags in the program are required and even if you have the flags but not the send command it still doesn't run the main because those flags are tied to the send command. I set up the switch statement in preparation for potentially implementing the latency profile of the server. In short, the default case will never be reached because of the way the flags and command are set up